components/Starknet/modules/architecture_and_concepts/pages/Network_Architecture/Fees/fee-mechanism.adoc

```diff
@@ -51,11 +51,16 @@ Let's go over a concrete example first. For example, imagine the following trace
 
 |===
 
-The proof will be closed and sent to L1 when any of these components becomes full. It's important to realize that the division to builtins must be predetermined! We can't decide on the fly to have proof with 20,000,001 Pedersen and nothing else.
+The proof will be closed and sent to L1 when any of these components becomes full.
 
-Suppose, for example, that a transaction uses 10,000 Cairo steps and 500 Pedersen hashes. We can squeeze at most 40,000 such transactions into our hypothetical trace (20,000,000/500). Therefore, its gas price will be correlated with 1/40,000 of the cost of submitting proof.
+[IMPORTANT]
+====
+The allocation of resources among built-in operations must be predetermined; it's not possible to decide post-execution to include only 20,000,001 Pedersen hashes without additional components.
+
+This safeguards fairness and prevents manipulation, ensuring integrity in proof generation and fee determination.
 
-Notice we completely ignored the number of Cairo steps in this transaction performance estimation, as it is not the limiting factor (since 500,000,000/10,000 > 20,000,000/500). With this example in mind, we can now formulate the exact fee associated with L2 computation.
+Notice we completely ignored the number of Cairo steps in this transaction performance estimation, as it is not the limiting factor.
```

Comment:
This should be outside of [IMPORTANT] along with the deleted paragraph (why was it deleted in the first place?):

> Suppose, for example, that a transaction uses 10,000 Cairo steps and 500 Pedersen hashes. We can squeeze at most 40,000 such transactions into our hypothetical trace (20,000,000/500). Therefore, its gas price will be correlated with 1/40,000 of the cost of submitting proof.

I.e., this should be:

> The proof will be closed and sent to L1 when any of these components becomes full.
> Suppose, for example, that a transaction uses 10,000 Cairo steps and 500 Pedersen hashes. We can squeeze at most 40,000 such transactions into our hypothetical trace (20,000,000/500). Therefore, its gas price will be correlated with 1/40,000 of the cost of submitting proof.
> Notice we completely ignored the number of Cairo steps in this transaction performance estimation, as it is not the limiting factor.
> 
> [IMPORTANT]
> The allocation of resources among built-in operations must be predetermined; it's not possible to decide post-execution to include only 20,000,001 Pedersen hashes without additional components.
> This safeguards fairness and prevents manipulation, ensuring integrity in proof generation and fee determination.

WDYT?